src/web3/strategy/strategies/controller.bdaoMembership.ts

```diff
@@ -0,0 +1,124 @@
+import apiConfig from './CONFIG';
+import { ERC20BalanceOfDto, ERC20TokenBalances, TokenList } from '../../token-vote/evm/evm.dtos';
+import { ethers } from 'ethers';
+import { strategyTypes } from '../../../common/constants';
+import { ResultTransformerParams, StrategyUtils } from '../strategy.types';
+
+const conf = {
+    api_tag: apiConfig.API_TAG,
+    api_url_base: apiConfig.API_TAG.toLowerCase(),
+    // modify to match your startegy setting in CONFIG.ts
+    name: apiConfig.STRATEGY_BANKLESS_DAO_MEMBERSHIP,
+    strategy_type: strategyTypes.STRATEGY_TYPE_TOKEN_GATED,
+    description: '1 vote if any of your wallets has >= 35k BANK',
+};
+
+// do not modify
+@ApiTags(conf.api_tag)
+@Controller(conf.api_url_base)
+
+// modify: rename class
+export class BdaoMembershipStrategy extends StrategyBaseController implements OnApplicationBootstrap {
+
+    // do not modify
+    constructor(
+        protected strategyService: StrategyBaseService,
+        protected strategyMongoService: StrategyMongoService,
+    ) {
+        super(strategyService, strategyMongoService);
+    }
+
+    // modify: implement strategy here
+    async strategy(strategyUtils: StrategyUtils) {
+
+        const polygonBlockHeight = strategyUtils.strategyRequest.block_height.find((blockHeight) => blockHeight.chain_id === '137').block;
+        const mainnetBlockHeight = strategyUtils.strategyRequest.block_height.find((blockHeight) => blockHeight.chain_id === '1').block;
+
+        strategyUtils.logger.debug(`blockHeights: ${JSON.stringify(strategyUtils.strategyRequest.block_height)}`);
+
+        strategyUtils.logger.debug(`equivalentBlockHeight: ${polygonBlockHeight}`);
+
+        const banklessTokenMain: ERC20BalanceOfDto = {
+            contractAddress: '0x2d94aa3e47d9d5024503ca8491fce9a2fb4da198',
+            chain_id: 1,
+            block_height: mainnetBlockHeight,
+        };
+
+        const banklessTokenPolygon: ERC20BalanceOfDto = {
+            contractAddress: '0xDB7Cb471dd0b49b29CAB4a1C14d070f27216a0Ab',
+            chain_id: 137,
+            block_height: polygonBlockHeight,
+        };
+
+        let tokens: TokenList;
+
+        if (banklessTokenPolygon.block_height) {
+            tokens = { tokens: [banklessTokenMain, banklessTokenPolygon] };
+
+        } else {
+            tokens = { tokens: [banklessTokenMain] };
+
+        }
+
+        strategyUtils.logger.debug(`getting balances for token list: ${JSON.stringify(tokens)}`);
+
+        return await strategyUtils.evmService.getErc20TokenBalances(strategyUtils.strategyRequest.account_id, tokens);
+    }
+
+    // transform strategy result, or use to chain strategies
+    responseTransformer(resultTransformerParams: ResultTransformerParams): string {
+        let votingPower = ethers.BigNumber.from('0');
+        const MEMBERSHIP_THRESHOLD = '35000000000000000000000';
+
+
+        for (const token of (resultTransformerParams.strategyResult as ERC20TokenBalances).tokenBalances) {
```

Comment:
so this is adding Mainnet and Polygon of single address

Reply:
because when User votes with Wallet A and Wallet B, the strategy is run twice.

Reply:
Yep